src/AdoPat/IPatClient.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// An abstraction over common operations with the Azure DevOps PAT Lifecycle Management REST API.
+    /// </summary>
+    public interface IPatClient
+    {
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">The PAT creation request.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        Task<PatToken> CreatePatAsync(PatTokenCreateRequest patTokenCreateRequest, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets all active PATs.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A mapping of authorization ID to active PATs.</returns>
+        Task<IDictionary<Guid, PatToken>> GetActivePatsAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Creates a new PAT with a new 'valid to' date by replicating an existing one.
+        /// </summary>
+        /// <param name="patToken">An existing <see cref="PatToken"/>.</param>
+        /// <param name="validTo">The new expiration date for the regenerated token.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatToken"/>.</returns>
+        Task<PatToken> RegeneratePatAsync(PatToken patToken, DateTime validTo, CancellationToken cancellationToken = default);

```

Comment:
Should we have a "Re-<create>" method at all? This actually seems confusing to me since we explicitly are NOT regenerating existing PATs, we are always making new PATs. I think this should be also named `CreateAsync` and it's an overload. We can make a new PAT by either taking a new request params, or an existing pat, to queue off of.

Reply:
This is documented in the [`PatClient`](https://github.com/AzureAD/microsoft-authentication-cli/blob/93e2778c0eefdec3d9c9f7665b422c73b591a930/src/AdoPat/PatClient.cs#L102-L114) code. We **are** regenerating a PAT according to the Azure DevOps description of that logical operation. That operation is a combination of API calls and is consistent with the UI that a user would see when managing their own PATs.

Reply:
That said, I'm happy to drop `Pat` from this method name and just go with `RegenerateAsync`, but I think we need to keep the `Regenerate` to be consistent with Azure DevOps.